tests/framework/db/ActiveRecordTest.php

```diff
@@ -1439,4 +1439,20 @@ public function testUnlinkAllOnConditionViaTable()
         // Make sure that only links were removed, the items were not removed
         $this->assertEquals(3, $itemClass::find()->where(['category_id' => 2])->count());
     }
+
+    /**
+     * https://github.com/yiisoft/yii2/pull/13891
+     */
+    public function testIndexByAfterLoadingRelations()
+    {
+        $orderClass = $this->getOrderClass();
+
+        $this->assertNotEmpty($orderClass::find()->with('customer')->indexBy(function($order) {
+            return isset($order->product->id) ? $order->product->id : null;
+        })->all());
+
+        $this->assertNotEmpty($orderClass::find()->with('customer')->indexBy('product.id')->all());
```

Comment:
maybe here you should call https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.assertArrayHasKey